modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/AndesMessageMetadata.java

```diff
@@ -315,15 +337,40 @@ private void parseMetaData() {
         buf = buf.slice();
         MessageMetaDataType type = MessageMetaDataType.values()[originalMetadata[0]];
         metaDataType = type;
-        StorableMessageMetaData original_mdt = type.getFactory()
-                .createMetaData(buf);
+        StorableMessageMetaData originalMessageMetadata = type.getFactory().createMetaData(buf);
+
+        byte[] underlying;
+        //TODO need to implement factory pattern here
+        if (type.equals(MessageMetaDataType.META_DATA_MQTT)) {
```

Comment:
Shall we flip the 'equals parameters as "MessageMetaDataType.META_DATA_MQTT.equals(type)"